BSP/Simulator/DataGeneration/ltc6811.py

```diff
@@ -0,0 +1,647 @@
+        '''
+        self.curr_cmd_code = new_cmd_code
+        # Handle cmd cases such as MD, ST, etc.
+        if data == None:
+            data = [0] * LTC6811_NUM_REG  # Initialize as empty
+        self.rx_data = data.copy()
+    
+
+    def update(self):
+        '''
+        @brief  Updates the LTC6811 object with the new command that was set in stage_update.
+        '''
+
+        # command_codes is nested dictionary
+        # {cmd_key : {code, handler}}
+        # cmd_key is the string representation of the code for readability
+        # code is the number value of the command
+        # handler is the function to call to handle its respective command
+        for cmd_key in command_codes:
+            if self.curr_cmd_code == command_codes[cmd_key]['code']:
+                cmd_handler = command_codes[cmd_key]['handler']
+                
+                # Execute only if handler exists
+                if cmd_handler is not None:
+                    cmd_handler(self)
+
+
+    def set_voltage(self, voltages):
+        self.voltages = voltages.copy()
+    
+    def set_temperatures(self, temperatures_cel):
+        temp_idx = 0
+
+        # Separate the two temperature sensors per module into separate LTC1380s
+        # [a,b] where a goes to mux1 and b goes to mux2
+        for a, b in temperatures_cel:
+            self.ltc1380s[mux_idxs['SIM_LTC1380_MUX1']].temperatures_cel[temp_idx] = a
+            self.ltc1380s[mux_idxs['SIM_LTC1380_MUX2']].temperatures_cel[temp_idx] = b
+            temp_idx += 1
+
+    def set_wires(self, wires):
+        '''
+        @brief  updates array of wires
+        @param  wires : list of wires, each index corresponds to batt module.
+                        1 for closed, 0 for open
+        '''
+        self.wires = wires.copy()
+
+    def set_single_wire(self, idx, isconnected):
+        '''
+        @brief  sets a single wire to open or closed
+        @param  idx : batt module 0-indexed
+        @param  isconnected : 1 for not open (closed), 1 for open
+        '''
+        self.wires[idx] = isconnected
+
+    def set_voltage_pullup(self, pu_volt):
+        self.pullup_volt = pu_volt.copy()
+
+    def set_voltage_pulldown(self, pd_volt):
+        self.pulldown_volt = pd_volt.copy()
+
+    def set_muxes(self, en, mux_addr, mux_sel):
+        for ltc1380 in self.ltc1380s:
+            ltc1380.update(en, mux_addr, mux_sel)
+
+    def update_gpios(self):
+        temperature_mcel = -1
+        for ltc1380 in self.ltc1380s:
+            value = ltc1380.get_temperature()
+            if value != -1:
+                # Valid
+                temperature_mcel = value
+
+        if temperature_mcel == -1:
+            logging.warning('update_gpios')
+        
+        # Convert GPIO temperature to m
+        celcius_float = temperature_mcel / 1000
+        temperature_mV = 2230.8 - (13.582 * (celcius_float - 30)) - (0.00433 * (celcius_float - 30) ** 2)
```

Comment:
Where do these numbers come from?

Reply:
This is from the LMT87 temperature sensor datasheet. I did not write this, this from taken from Temperature.c